scripts/vale/vale.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const chalk = require( 'chalk' );
+const upath = require( 'upath' );
+const minimist = require( 'minimist' );
+const { globSync } = require( 'glob' );
+const { spawn } = require( 'child_process' );
+
+const CKEDITOR5_ROOT = upath.join( __dirname, '..', '..' );
+
+const minimistOptions = {
+	string: [
+		'directory',
+		'files'
+	],
+	alias: {
+		d: 'directory',
+		f: 'files'
+	},
+	default: {
+		directory: '',
+		files: ''
+	}
+};
+
+const spawnOptions = {
+	cwd: CKEDITOR5_ROOT,
+	shell: true
+};
+
+const globOptions = {
+	cwd: CKEDITOR5_ROOT,
+	ignore: '**/node_modules/**'
+};
+
+const itemsToCheckInsidePackages = [
+	'README.md',
+	'docs'
+];
+
+const defaultPatterns = [
+	'CHANGELOG.md',
+	'CONTRIBUTING.md',
+	'LICENSE.md',
+	'README.md',
+	'SECURITY.md',
+	'docs',
+	'external/ckeditor5-commercial/docs',
+	...getPathsFromPackages( 'packages' ),
+	...getPathsFromPackages( 'external/ckeditor5-commercial/packages' )
+];
+
+main();
+
+function main() {
+	const args = minimist( process.argv.slice( 2 ), minimistOptions );
+	const patterns = args.directory ?
+		[ `**/${ args.directory }/**/*.md` ] :
+		args.files.split( ',' ).filter( Boolean );
+	const valeArgs = [ 'run', 'docs:vale' ];
+
+	if ( patterns.length ) {
+		const files = patterns.flatMap( pattern => globSync( pattern, globOptions ) );
```

Comment:
`glob` supports array of patterns:

```suggestion
		const files = globSync( patterns, globOptions );
```